kb/development/python-environment.md

```diff
@@ -0,0 +1,50 @@
+# Setting up a Python environment for Mbed TLS
+
+## Python and package versions
+
+Each branch of Mbed TLS has different minimum requirements on Python. See `README.md` in the branch. Any Python version that's more recent than the minimum should work, however as of November 2022 this is not well tested by the CI so unintended incompatibilities can happen.
+
+Some of the scripts used by Mbed TLS users and developers require third-party packages. Those have version constraints as well. The constraints are documented via `*.requirements.txt` files in the [`scripts` directory](https://github.com/Mbed-TLS/mbedtls/tree/development/scripts).
+
+We run [mypy](http://mypy-lang.org/) and [pylint](https://pylint.pycqa.org/en/latest/) to check Python scripts. It is common that newer versions of Pylint reject our code, because some checks have become stricter or new checks have been added.
+
+## Set up a Python package environment
+
+### Quick setup
+
+Initial setup:
+
+```sh
+python3 -m venv ~/venv-mbedtls-development
+~/venv-mbedtls-development/bin/python scripts/min_requirements.py
+```
+
+Day-to-day work: start a shell and issue this command:
+```sh
+. ~/venv-mbedtls-development/bin/activate
```

Comment:
Should you unset `PYTHONPATH` as well? This can cause disruption sometimes.

Reply:
I don't think that should be in the basic instructions, but might worth mentioning in a remark.

Reply:
With regards to PYTHONPATH I believe nowadays venv is unsetting `PYTHONHOME`. 

But there is nothing stopping users from adding an extra unset for it in ~/venv-mbedtls-development/bin/activate

It falls outside of the scope of this document though. If needed we could have a remark in line of 

> If required `$PYTHONPATH` and `$PYTHONHOME` can be configured or unset by modifying `~/venv-mbedtls-development/bin/activate`